client/clip_client/client.py

```diff
@@ -159,17 +159,30 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
             if isinstance(c, str):
                 _mime = mimetypes.guess_type(c)[0]
                 if _mime and _mime.startswith('image'):
-                    yield Document(
-                        tags={'__created_by_CAS__': True}, uri=c
-                    ).load_uri_to_blob()
+                    try:
+                        yield Document(
+                            tags={'__created_by_CAS__': True}, uri=c
+                        ).load_uri_to_blob()
+                    except Exception:
+                        raise ValueError(
+                            f'Failed to load image {c}. Maybe the URI is broken?'
+                        )
                 else:
                     yield Document(tags={'__created_by_CAS__': True}, text=c)
             elif isinstance(c, Document):
                 if c.content_type in ('text', 'blob'):
                     yield c
                 elif not c.blob and c.uri:
-                    c.load_uri_to_blob()
-                    yield c
+                    try:
+                        c.load_uri_to_blob()
+                    except Exception:
+                        warnings.warn(
+                            f'Failed to load image doc with ID {c.id}.'
+                            'Maybe the URI is broken?'
+                            'You will get empty embedding for this doc.'
+                        )
+                    finally:
+                        yield c
```

Comment:
at the client side, we should raise an exception to stop the encoding.

Reply:
but what about ppl using clip_client who have encoded 98% of a large da and suddenly broke?
if we do not catch this, we can never get empty embeddings

Reply:
ah, yes you are right. But please ignore the broken doc, rather than pass to server in `finally`

Reply:
i can but if we simply ignore them, they may disappear in the result. not 100 sure, double checking